<a href="https://colab.research.google.com/github/Pranav-22-aditya/Verzeo-Projects/blob/main/ML_MINOR_JAN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from sklearn import metrics
from sklearn.metrics import confusion_matrix

In [ ]:
df=pd.read_csv("/content/drive/MyDrive/Verzeo_python/Iris.csv")

In [ ]:
df


Id  SepalLengthCm  ...  PetalWidthCm         Species
0      1            5.1  ...           0.2     Iris-setosa
1      2            4.9  ...           0.2     Iris-setosa
2      3            4.7  ...           0.2     Iris-setosa
3      4            4.6  ...           0.2     Iris-setosa
4      5            5.0  ...           0.2     Iris-setosa
..   ...            ...  ...           ...             ...
145  146            6.7  ...           2.3  Iris-virginica
146  147            6.3  ...           1.9  Iris-virginica
147  148            6.5  ...           2.0  Iris-virginica
148  149            6.2  ...           2.3  Iris-virginica
149  150            5.9  ...           1.8  Iris-virginica

[150 rows x 6 columns]

checking the data set 

In [ ]:
df.isnull().sum()


Id               0
SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

since no null value. thus we don't need to drop any row or column.

In [ ]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()

In [ ]:
#df1=pd.get_dummies(df)
#df1

In [ ]:
y = df.iloc[:,-1 ].values

In [ ]:
y

array(['Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-versicolor', 'Iris-versicolor',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-versicolor',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-versic

In [ ]:
y = le.fit_transform(y)
y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [ ]:
x = df.iloc[:, 1:5].values
x

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
x_train,x_test,y_train,y_test = train_test_split(x,y, test_size = 0.2, random_state=1, stratify = y)


In [ ]:
x_test

array([[7.3, 2.9, 6.3, 1.8],
       [4.9, 3.1, 1.5, 0.1],
       [5.1, 2.5, 3. , 1.1],
       [4.8, 3.4, 1.6, 0.2],
       [5. , 3.5, 1.6, 0.6],
       [5.1, 3.5, 1.4, 0.2],
       [6.2, 3.4, 5.4, 2.3],
       [6.4, 2.7, 5.3, 1.9],
       [5.6, 2.8, 4.9, 2. ],
       [6.8, 2.8, 4.8, 1.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.5, 2.3, 4. , 1.3],
       [6.8, 3. , 5.5, 2.1],
       [6. , 2.2, 4. , 1. ],
       [5.7, 2.5, 5. , 2. ],
       [5.7, 4.4, 1.5, 0.4],
       [7.1, 3. , 5.9, 2.1],
       [6.1, 2.8, 4. , 1.3],
       [4.9, 2.4, 3.3, 1. ],
       [6.1, 3. , 4.9, 1.8],
       [6.4, 2.9, 4.3, 1.3],
       [5.6, 3. , 4.5, 1.5],
       [4.9, 3.1, 1.5, 0.1],
       [4.4, 2.9, 1.4, 0.2],
       [6.5, 3. , 5.2, 2. ],
       [4.9, 2.5, 4.5, 1.7],
       [5.4, 3.9, 1.7, 0.4],
       [4.8, 3. , 1.4, 0.1],
       [6.3, 3.3, 4.7, 1.6],
       [6.5, 2.8, 4.6, 1.5]])

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors=4, metric='euclidean')
classifier.fit(x_train,y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='euclidean',
                     metric_params=None, n_jobs=None, n_neighbors=4, p=2,
                     weights='uniform')

In [ ]:
y_test

array([2, 0, 1, 0, 0, 0, 2, 2, 2, 1, 0, 1, 2, 1, 2, 0, 2, 1, 1, 2, 1, 1,
       0, 0, 2, 2, 0, 0, 1, 1])

In [ ]:
pred_y = classifier.predict(x_test)

In [ ]:
pred_y

array([2, 0, 1, 0, 0, 0, 2, 2, 2, 1, 0, 1, 2, 1, 2, 0, 2, 1, 1, 2, 1, 1,
       0, 0, 2, 1, 0, 0, 1, 1])

In [ ]:
from sklearn.metrics import confusion_matrix

In [ ]:
cm = confusion_matrix(y_test, pred_y)

In [ ]:
cm

array([[10,  0,  0],
       [ 0, 10,  0],
       [ 0,  1,  9]])

In [ ]:
score = classifier.score(x_test, y_test)
score

0.9666666666666667

thus the accuracy by knn method is 96.67%.

predicting some new data 

In [ ]:
new =classifier.predict([[  4.7,3.0	,1.3,	0.2 ]])
new[0]

0

In [ ]:
newx=new[0]
if (newx== 0): 
  print("Iris-setosa")
elif (newx == 1):
  print("Iris-versicolor")  
else :
  print("Iris-virginica")  

Iris-setosa


In [ ]:
#now trying svm classifier

In [ ]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
x_train = sc.fit_transform(x_train)
x_test = sc.transform(x_test)

in above step we did Feature Scaling.

In [ ]:
print(x_test)

[[ 1.70737111 -0.39101277  1.40726524  0.77483369]
 [-1.15647793  0.08777838 -1.28391928 -1.43279303]
 [-0.91782384 -1.34859505 -0.44292412 -0.13418908]
 [-1.27580497  0.80596509 -1.22785293 -1.30293264]
 [-1.03715089  1.04536066 -1.22785293 -0.78349105]
 [-0.91782384  1.04536066 -1.33998562 -1.30293264]
 [ 0.39477364  0.80596509  0.90266814  1.42413567]
 [ 0.63342772 -0.86980391  0.8466018   0.90469409]
 [-0.32118863 -0.63040834  0.62233642  1.03455449]
 [ 1.1107359  -0.63040834  0.56627007  0.25539211]
 [-0.55984271  2.00294294 -1.39605197 -1.04321185]
 [-0.44051567 -1.82738619  0.11773932  0.12553172]
 [ 1.1107359  -0.15161719  0.95873448  1.16441488]
 [ 0.15611955 -2.06678176  0.11773932 -0.26404947]
 [-0.20186158 -1.34859505  0.67840276  1.03455449]
 [-0.20186158  3.1999208  -1.28391928 -1.04321185]
 [ 1.46871703 -0.15161719  1.18299986  1.16441488]
 [ 0.27544659 -0.63040834  0.11773932  0.12553172]
 [-1.15647793 -1.58799062 -0.27472509 -0.26404947]
 [ 0.27544659 -0.15161719  0.62

In [ ]:
print(x_train)

[[-0.91782384  1.5241518  -1.28391928 -1.04321185]
 [-1.51445906  0.32717395 -1.33998562 -1.30293264]
 [ 1.23006294  0.08777838  0.73446911  1.42413567]
 [-0.44051567 -1.10919948  0.3420047  -0.00432868]
 [-0.32118863 -0.39101277 -0.10652605  0.12553172]
 [ 0.15611955  0.80596509  0.39807104  0.5151129 ]
 [ 0.51410068 -1.82738619  0.3420047   0.12553172]
 [ 0.15611955 -2.06678176  0.67840276  0.38525251]
 [ 0.51410068  0.80596509  1.01480083  1.55399607]
 [ 1.23006294  0.08777838  0.62233642  0.38525251]
 [ 0.99140885 -0.15161719  0.79053545  1.42413567]
 [-1.03715089  0.32717395 -1.45211831 -1.30293264]
 [-0.55984271 -0.15161719  0.39807104  0.38525251]
 [ 0.15611955 -0.86980391  0.73446911  0.5151129 ]
 [ 0.51410068 -0.39101277  1.01480083  0.77483369]
 [-1.27580497  0.80596509 -1.0596539  -1.30293264]
 [ 0.99140885 -1.34859505  1.12693352  0.77483369]
 [ 0.39477364 -2.06678176  0.39807104  0.38525251]
 [-0.32118863 -0.86980391  0.22987201  0.12553172]
 [-1.03715089  0.80596509 -1.28

In [ ]:
#Training the SVM model on the Training set

In [ ]:
from sklearn.svm import SVC
classifier = SVC(kernel = 'linear', random_state = 0)
classifier.fit(x_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='linear',
    max_iter=-1, probability=False, random_state=0, shrinking=True, tol=0.001,
    verbose=False)

In [ ]:
x_train

array([[-0.91782384,  1.5241518 , -1.28391928, -1.04321185],
       [-1.51445906,  0.32717395, -1.33998562, -1.30293264],
       [ 1.23006294,  0.08777838,  0.73446911,  1.42413567],
       [-0.44051567, -1.10919948,  0.3420047 , -0.00432868],
       [-0.32118863, -0.39101277, -0.10652605,  0.12553172],
       [ 0.15611955,  0.80596509,  0.39807104,  0.5151129 ],
       [ 0.51410068, -1.82738619,  0.3420047 ,  0.12553172],
       [ 0.15611955, -2.06678176,  0.67840276,  0.38525251],
       [ 0.51410068,  0.80596509,  1.01480083,  1.55399607],
       [ 1.23006294,  0.08777838,  0.62233642,  0.38525251],
       [ 0.99140885, -0.15161719,  0.79053545,  1.42413567],
       [-1.03715089,  0.32717395, -1.45211831, -1.30293264],
       [-0.55984271, -0.15161719,  0.39807104,  0.38525251],
       [ 0.15611955, -0.86980391,  0.73446911,  0.5151129 ],
       [ 0.51410068, -0.39101277,  1.01480083,  0.77483369],
       [-1.27580497,  0.80596509, -1.0596539 , -1.30293264],
       [ 0.99140885, -1.

In [ ]:
x_test

array([[ 1.70737111, -0.39101277,  1.40726524,  0.77483369],
       [-1.15647793,  0.08777838, -1.28391928, -1.43279303],
       [-0.91782384, -1.34859505, -0.44292412, -0.13418908],
       [-1.27580497,  0.80596509, -1.22785293, -1.30293264],
       [-1.03715089,  1.04536066, -1.22785293, -0.78349105],
       [-0.91782384,  1.04536066, -1.33998562, -1.30293264],
       [ 0.39477364,  0.80596509,  0.90266814,  1.42413567],
       [ 0.63342772, -0.86980391,  0.8466018 ,  0.90469409],
       [-0.32118863, -0.63040834,  0.62233642,  1.03455449],
       [ 1.1107359 , -0.63040834,  0.56627007,  0.25539211],
       [-0.55984271,  2.00294294, -1.39605197, -1.04321185],
       [-0.44051567, -1.82738619,  0.11773932,  0.12553172],
       [ 1.1107359 , -0.15161719,  0.95873448,  1.16441488],
       [ 0.15611955, -2.06678176,  0.11773932, -0.26404947],
       [-0.20186158, -1.34859505,  0.67840276,  1.03455449],
       [-0.20186158,  3.1999208 , -1.28391928, -1.04321185],
       [ 1.46871703, -0.

In [ ]:
#train the model with svm

In [ ]:
from sklearn.svm import SVC
classifier = SVC(kernel = 'linear', random_state = 0)
classifier.fit(x_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='linear',
    max_iter=-1, probability=False, random_state=0, shrinking=True, tol=0.001,
    verbose=False)

In [ ]:
y_pred = classifier.predict(x_test)

In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test, y_pred)

[[10  0  0]
 [ 0 10  0]
 [ 0  1  9]]


0.9666666666666667

the accuracy with svm is also found to be 96.67%.